src/components/Checkout/CheckoutAPICalls.tsx

```diff
@@ -2,9 +2,9 @@ import { getRole } from '../contexts/UserContext';
 import { CheckoutItem, ClientPrincipal } from '../../types/interfaces';
 import { ENDPOINTS, HEADERS } from '../../types/constants';
 
-export async function processWelcomeBasket(user: ClientPrincipal | null, currentUserId: number, checkoutItems: CheckoutItem[]) {
+export async function processWelcomeBasket(user: ClientPrincipal | null, currentUserId: number, checkoutItems: CheckoutItem[], buildingCode: string) {
   HEADERS['X-MS-API-ROLE'] = getRole(user);
-  const response = await fetch(ENDPOINTS.CHECKOUT_WELCOME_BASKET, {
+  const response = await fetch(`${ENDPOINTS.CHECKOUT_WELCOME_BASKET}?buildingCode=${buildingCode}`, {
```

Comment:
Can we move the buildingCode into the JSON body, like the rest? Like this

user_id: currentUserId,
mattress_size: checkoutItems[0].id,
building_code: buildingCode,